release/release_tests.yaml

```diff
@@ -3676,7 +3676,7 @@
     script: pytest checkpointing_tests/test_learner_group_checkpointing.py
 
     wait_for_nodes:
-      num_nodes: 3
+      num_nodes: 2
```

Comment:
why is this node number change required?

Reply:
A previous PR reduces the number of node to 2 https://github.com/ray-project/ray/pull/36929, so this needs to update as well